test/content/support.js

```diff
@@ -241,4 +241,26 @@ function resetDB() {
 	}).then(function() {
 		return Zotero.Schema.schemaUpdatePromise;
 	});
-}
\ No newline at end of file
+}
+
+
+/**
+ * Imports an attachment and calls a function when it is ready.
+ * @param {string} filename - The filename to import (in data directory)
+ * @param {function} f - A function that takes on argument, the id of the item.
+ * @returns {Promise}
+ */
+function* withImportedAttachmentFromFile(filename, f) {
+	let testfile = getTestDataDirectory();
+	testfile.append(filename);
+	let insertedId = yield Zotero.Attachments.importFromFile({file: testfile});
+	return waitForItemEvent("refresh").then(function(ids) {
```

Comment:
You shouldn't need to wait for refresh event. `Zotero.Attachments.importFromFile` returns a promise for an itemID, which is resolved only after the item is added to the DB. I think the other places in the test suite where this us used do not return such promises. I'm guessing that you might not ever even see this event fire at this point (after `Zotero.Attachments.importFromFile` has already resolved above)